erc20/ds-token/README.md

```diff
@@ -0,0 +1,117 @@
+commit [`1a1a5b6`][version]:
+* [token.sol][token]
+* [base.sol][base]
+
+We formally verified the full functional correctness of the following ERC20 functions:
+
+* [`totalSupply`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L27) - inherited in DSToken from DSTokenBase 
+* [`balanceOf`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L30) - inherited in DSToken from DSTokenBase 
+* [`allowance`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L33) - inherited in DSToken from DSTokenBase 
+* [`approve`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L35) which extends [`approve`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L61) from DSTokenBase
+* [`transfer`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L27) which extends [`transfer`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L37) from DSTokenBase
+* [`transferFrom`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/token.sol#L30) which extends [`transferFrom`](https://github.com/dapphub/ds-token/blob/bb98ff4841e82137efc9690b9e1cf77e85fdd0bb/src/base.sol#L48) from DSTokenBase
+
+## Verification Artifacts
+
+### Solidity Source Code and Compiled EVM Bytecode
+
+We took the [token source code][token], [base token source code][base] and the additional libraries [`DSAuth`](https://github.com/dapphub/ds-auth/tree/ce62361903fbf085c94c82c885c7ae393c95749e), [`DSMath`](https://github.com/dapphub/ds-math/tree/228177aa957f87ac6f9bb6a97b64a7912441323a), [`DSStop`](https://github.com/dapphub/ds-stop/tree/32fffeb2142cb34b71d8a7914f784244b44d1ede) and [`DSNote`](https://github.com/dapphub/ds-note/tree/8bc2e258055516dde772013212893266bc9d6449), and compiled it to the EVM bytecode using solc compiler (of the version `0.4.18+commit.9cf6e910.Linux.g++`).
+
+The EVM (runtime) bytecode generated by the solc compiler is the following:
+
+* [ds-token.bytes](ds-token.bytes)
+
+The (annotated) EVM assembly disassembled from the bytecode is the following:
+
+* [ds-token.evm](ds-token.evm)
+
+### Mechanized Specifications and Proofs
+
+Due to its deviation from [ERC20-K], we could not verify the DSToken ERC20 token against the original [ERC20-EVM] specification. In order to show that it is "correct" w.r.t. ERC20-K (thus ERC20-EVM) modulo the deviation, we modified the specification to capture the deviation and successfully verified it against the modified ERC20-EVM specification. Below are the changes made to the original ERC20-EVM specification:
+
+* To capture the exception in `transfer`, `transferFrom` and `approve`, we added the following variable in `storage`
+```
+    #hashedLocation({COMPILER}, {_STOPPED}, .IntList) |-> STOPPED
+```
```

Comment:
be careful with the leading white spaces, which *matters* in the markdown file.

the three backticks (```) should starts with the four white spaces.